### PR TITLE
Adds the individual algorithm for utxo selection, which allows

### PR DESCRIPTION
**Pull request by thecodefactory**
Adds the individual algorithm for utxo selection, which allows selection of all utxos available that can individually satisfy the amount requested.

### CODE REVIEW

File: src/wallet/select_outputs.cpp [hunk 1]
```diff
@@ -92,5 +84,38 @@ void select_outputs::select(points_info& out, output_info::list unspent,
...
+    switch(option)
+    {
+        case algorithm::greedy:
+            greedy_select(out, unspent, minimum_value);
+            break;
+
+        case algorithm::individual:
+            individual_select(out, unspent, minimum_value);
+            break;
+    }
```
Comment (evoskuil) [inline src/wallet/select_outputs.cpp]: Include `default` in switch.

Comment (evoskuil) [inline src/wallet/select_outputs.cpp]: Include `default` in switch.

File: src/wallet/select_outputs.cpp [hunk 2]
```diff
@@ -92,5 +84,38 @@ void select_outputs::select(points_info& out, output_info::list unspent,
...
+    {
+        if (cur_unspent.value > minimum_value)
+        {
+            out.points.push_back(cur_unspent.point);
+        }
+    }
+}
+
+void select_outputs::select(points_info& out, output_info::list unspent,
+    const uint64_t minimum_value, const algorithm option)
```
Comment (evoskuil) [inline src/wallet/select_outputs.cpp]: Algorithm is copied parameter (no `const`).

File: src/wallet/select_outputs.cpp [hunk 3]
```diff
@@ -92,5 +84,38 @@ void select_outputs::select(points_info& out, output_info::list unspent,
     out.points.clear();
 }
 
-} // namespace wallet
-} // namespace libbitcoin
+static void individual_select(points_info& out, const output_info::list unspent,
+    const uint64_t minimum_value)
+{
+    for (const auto& cur_unspent : unspent)
```
Comment (evoskuil) [inline src/wallet/select_outputs.cpp]: Avoid abbreviations (`cur_`).

No space prior to `:`.

File: include/bitcoin/bitcoin/wallet/select_outputs.hpp [hunk 4]
```diff
@@ -29,16 +29,29 @@ namespace wallet {
...
+        greedy,
+        individual
     };
 
     /// Select optimal outpoints for a spend from unspent outputs list.
     /// Return includes the amount of change remaining from the spend.
     static void select(chain::points_info& out,
-        chain::output_info::list unspent, uint64_t minimum_value,
-        algorithm option=algorithm::greedy);
+        chain::output_info::list unspent, const uint64_t minimum_value,
```
Comment (evoskuil) [inline include/bitcoin/bitcoin/wallet/select_outputs.hpp]: `chain::output_info::list unspent` should be `const &` (same below).

Avoid `const` in by-value parameters (same below).

File: src/wallet/select_outputs.cpp [hunk 5]
```diff
@@ -92,5 +84,38 @@ void select_outputs::select(points_info& out, output_info::list unspent,
...
+    {
+        if (cur_unspent.value > minimum_value)
+        {
+            out.points.push_back(cur_unspent.point);
+        }
+    }
```
Comment (evoskuil) [inline src/wallet/select_outputs.cpp]: No braces on simple single line scopes (`for` and `if` in this case).

File: src/wallet/select_outputs.cpp [hunk 6]
```diff
@@ -92,5 +84,38 @@ void select_outputs::select(points_info& out, output_info::list unspent,
...
 }
 
-} // namespace wallet
-} // namespace libbitcoin
+static void individual_select(points_info& out, const output_info::list unspent,
+    const uint64_t minimum_value)
+{
+    for (const auto& cur_unspent : unspent)
+    {
+        if (cur_unspent.value > minimum_value)
```
Comment (evoskuil) [inline src/wallet/select_outputs.cpp]: Why not `>=` ?

**Comment by thecodefactory** [comment]
Updated.  Haven't had time yet to add a contrived offline test for this, but a real test is used in explorer.

**Pull request merged**